scipy/signal/_peak_finding.py

```diff
@@ -233,6 +234,733 @@ def argrelextrema(data, comparator, axis=0, order=1, mode='clip'):
+#
+#     Copyright (c) 2013 Marcos Duarte
+#
+#     Permission is hereby granted, free of charge, to any person
+#     obtaining a copy of this software and associated documentation
+#     files (the "Software"), to deal in the Software without
+#     restriction, including without limitation the rights to use,
+#     copy, modify, merge, publish, distribute, sublicense, and/or sell
+#     copies of the Software, and to permit persons to whom the
+#     Software is furnished to do so, subject to the following
+#     conditions:
+#
+#     The above copyright notice and this permission notice shall be
+#     included in all copies or substantial portions of the Software.
+#
+#     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+#     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+#     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+#     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+#     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+#     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+#     OTHER DEALINGS IN THE SOFTWARE.
+
+def _filter_by_peak_distance(peaks, priority, dmin):
+    """
+    Flag neighbouring peaks for removal to ensure a minimal distance.
+
+    Parameters
+    ----------
+    peaks : ndarray
+        Indices of peaks in `vector`.
+    priority : ndarray
+        An array with priorities matching `peaks` used to determine priority of
+        peaks. A peak with a higher priority value is kept over one with a lower
+        one.
+    dmin : number
+        Minimal distance that peaks must be spaced.
+
+    Returns
+    -------
+    keep : ndarray[bool]
+        A boolean mask matching `peaks` that evaluates to true for each peak
+        that must be removed in order to fullfill the requirement.
+
+    Notes
+    -----
+
+    .. versionadded:: 1.1.0
+    """
+    # Peaks are evaluated by priority (larger first)
+    eval_peaks = peaks[np.argsort(priority)][::-1]
+
+    # Flag peaks for deletion
+    del_flag = np.zeros(eval_peaks.size, dtype=bool)
+    for i in range(eval_peaks.size):
+        if not del_flag[i]:
+            # Flag peaks in intervall +-distance around current peak
+            del_flag |= (eval_peaks > eval_peaks[i] - dmin) \
+                        & (eval_peaks < eval_peaks[i] + dmin)
+            # Keep current peak
+            del_flag[i] = False
+
+    keep = ~del_flag[np.argsort(eval_peaks)]
+
+    return keep
+
+
+def find_peaks(x, height=None, threshold=None, distance=None,
+               prominence=None, width=None, wlen=None, rel_height=0.5):
+    """
+    Find a subset of peaks inside a signal.
+
+    This function takes a one-dimensional array and finds all local maxima by
+    simple comparision of neighbouring values. A subset of these peaks can be
+    selected with different filters described below.
+
```

Comment:
It would be good to have one or two sentences here about what this function's limitations are - e.g. for noisy signals the approach doesn't work too well, for that it's better to use `find_peaks_cwt`.

Reply:
Something like:

"Because these filters depend on the calculation of properties for each peak this function can be slow for noisy data that has many small peaks relative to its length. In this case it is recommended to smooth the signal before peak finding and / or reduce the number of peaks with fast filter options first."

This feels more like it belongs in the Notes section.

Reply:
Yeah, I should have thought of that. I definitely add this or something like it either here or in the Notes section. 👍